src/main/java/com/siit/hospital_manager/controller/AppointmentController.java

```diff
@@ -0,0 +1,146 @@
+            List<AppointmentDto> appointmentsList = appointmentService.findAllByDoctorUserName(principal.getName(),AppointmentStatus.CONFIRMED);
+            model.addAttribute("appointments", appointmentsList);
+            return "doctor/viewAllAppointments";
+        } else {
+            List<AppointmentDto> appointmentsList = appointmentService.findAll();
+            model.addAttribute("appointments", appointmentsList);
+            return "admin/viewAllAppointments";
+        }
+    }
+
+    @GetMapping("/findAllByPatientId/{Id}")
+    public String findAllByPatientId(Model model, @PathVariable("Id") Integer id, Authentication authentication, AppointmentStatus appointmentStatus) {
+        if (isAdmin(authentication)) {
+            List<AppointmentDto> appointmentsList = appointmentService.findAllByPatientId(id, AppointmentStatus.CONFIRMED);
+            model.addAttribute("appointments", appointmentsList);
+            return "admin/viewPatientAppointments";
+        }
+        return "index";
+    }
+
+    @GetMapping("/findAllByDoctorId/{Id}")
+    public String findAllByDoctorId(Model model, @PathVariable("Id") Integer id, Authentication authentication, AppointmentStatus appointmentStatus) {
+        if (isAdmin(authentication)) {
+            List<AppointmentDto> appointmentsList = appointmentService.findAllByDoctorId(id, AppointmentStatus.CONFIRMED);
+            model.addAttribute("appointments", appointmentsList);
+            return "admin/viewDoctorAppointments";
+        }
+        return "index";
+    }
+
+    @DeleteMapping("/{id}")
+    @ResponseStatus(HttpStatus.ACCEPTED)
+    public void deleteAppointmentById(@PathVariable Integer id, Principal principal, Authentication authentication) {
+        if (isPatient(authentication)) {
+            appointmentService.deleteAppointmentByIdAndPatient(id, principal.getName());
+        } else if (isDoctor(authentication)) {
+            appointmentService.deleteAppointmentByIdAndDoctor(id, principal.getName());
+        } else appointmentService.deleteAppointmentById(id);
+    }
+
+    @GetMapping("/create/{doctorId}")
+    public String createAppointment(@PathVariable("doctorId") Integer id, CreateAppointmentDto createAppointmentDto, Principal principal, Model model) {
+        model.addAttribute("createAppointmentDto", createAppointmentDto);
+        model.addAttribute("doctorId", id);
+        appointmentService.addDoctorToCreateAppointmentDto(createAppointmentDto, id);
+        appointmentService.addPatientToCreateAppointmentDto(createAppointmentDto, principal.getName());
+        return "appointment/create";
+    }
+
+    @PostMapping("/submit/{doctorId}")
+    public String submitAppointmentForm(@Valid CreateAppointmentDto createAppointmentDto, BindingResult bindingResult, @PathVariable("doctorId") Integer id, Principal principal, Model model) {
+        appointmentService.addDoctorToCreateAppointmentDto(createAppointmentDto, id);
```

Comment:
this is a getDoctorById and add to create appointment? if so, this logic should be in a service